infer/src/pulse/PulseFormula.ml

```diff
@@ -3535,7 +3622,9 @@ module Formula = struct
                                       Atom.eval ~is_neq_zero:(is_neq_zero phi)
                                         (NotEqual (Var var, Term.zero))
                                     in
-                                    and_normalized_atoms (phi, new_eqs) atoms >>| snd )
+                                    and_normalized_atoms (phi, new_eqs) atoms ~orig_atom:atoms
```

Comment:
this seems dubious: we are propagating equalities discovered during normalisation so we shouldn't need to add any new raw atoms at this point.